packages/nx/src/utils/create-package-json.ts

```diff
@@ -15,11 +30,66 @@ export function createPackageJson(
   projectName: string,
   graph: ProjectGraph,
   options: {
+    target?: string;
     root?: string;
     isProduction?: boolean;
-  } = {}
+  } = {},
+  additionalDependencies?: string[]
 ): PackageJson {
-  const npmDeps = findAllNpmDeps(projectName, graph);
+  let selfInputs: string[] = [];
+  let dependencyInputs: string[] = [];
+
+  if (options.target) {
+    const projectNode = graph.nodes[projectName];
```

Comment:
This can be outside of the `if` branch and fed to `findAllNpmDeps` instead of `projectName` so we don't do same lookup twice